src/treq/test/test_response.py

```diff
@@ -98,3 +99,18 @@ def test_history_notimplemented(self):
 
     if not skip_history:
         test_history_notimplemented.skip = "History supported."
+
+    def test_fail_for_status_ok(self):
+        original = FakeResponse(200, Headers(), body=[b''])
+        response = _Response(original, None)
+        d = response.fail_for_status()
+        d.addCallback(self.assertEqual, response)
+
+        return d
```

Comment:
The modern way to write this assertion is:

```
d = response.fail_for_status()
self.assertEqual(response, self.successResultOf(d))
```

This parallels the use of `failureResultOf` in the next test.